### PR TITLE
boot: tweak resealing with fde-setup hooks

### PR DESCRIPTION
**Pull request by mvo5**
This is a small followup on #9783 to tweak/improve the error handling and tests.

### CODE REVIEW

File: boot/seal_test.go [hunk 1]
```diff
@@ -1009,3 +1009,32 @@ func (s *sealSuite) TestResealKeyToModeenvWithFdeHookSad(c *C) {
 	c.Assert(err, IsNil)
 	c.Check(resealKeyToModeenvUsingFDESetupHookCalled, Equals, 1)
 }
+
+func (s *sealSuite) TestResealKeyToModeenvWithFdeHookVerySad(c *C) {
```
Comment (pedronis) [inline boot/seal_test.go:1013]: I think Ian asked also for a test where HasFDESetupHook fails? (of course is not called anymore, but that's how the old code was failing)

**Review by pedronis — APPROVED** [review]
seal.go and assets.go changes look good. a remark about test additions

File: boot/assets.go [hunk 2]
```diff
@@ -360,9 +360,10 @@ func TrustedAssetsUpdateObserverForModel(model *asserts.Model, gadgetDir string)
 	switch {
 	case err == nil:
 		trackTrustedAssets = true
-	case err == errSealingNoKeys:
-		// nothing
-	case err != nil && err != errSealingNoKeys:
+	case err == errNoSealedKeys:
+		// nothing to do
+	case err != nil:
```
Comment (anonymouse64) [inline boot/assets.go:365]: thanks for this

File: boot/seal_test.go (hunk 1 above)
Comment (anonymouse64) [reply boot/seal_test.go:1013]: yes indeed I would like to make sure that doesn't accidentally happen again :-)

**Review by anonymouse64 — COMMENTED** [review]
lgtm but I still would like to see the addtl test I requested :-D

File: boot/seal_test.go (hunk 1 above)
Comment (mvo5) [reply boot/seal_test.go:1021]: Oh, sorry, I misunderstood. I added something now, I hope I got it right.

File: boot/seal_test.go [hunk 3]
```diff
@@ -993,6 +993,14 @@ func (s *sealSuite) TestResealKeyToModeenvWithFdeHookSad(c *C) {
...
 	defer restore()
 
+	// TODO: this simulates that the hook is not available yet
+	//       because of e.g. seeding. Longer term there will be
+	//       more, see TODO in resealKeyToModeenvUsingFDESetupHookImpl
+	restore = boot.MockHasFDESetupHook(func() (bool, error) {
+		return false, fmt.Errorf("hook may not available yet because e.g. seeding")
+	})
+	defer restore()
+
```
Comment (anonymouse64) [inline boot/seal_test.go:1003]: perfect thanks for this

File: boot/seal_test.go [hunk 4]
```diff
@@ -993,6 +993,14 @@ func (s *sealSuite) TestResealKeyToModeenvWithFdeHookSad(c *C) {
 	})
 	defer restore()
 
+	// TODO: this simulates that the hook is not available yet
+	//       because of e.g. seeding. Longer term there will be
+	//       more, see TODO in resealKeyToModeenvUsingFDESetupHookImpl
+	restore = boot.MockHasFDESetupHook(func() (bool, error) {
+		return false, fmt.Errorf("hook may not available yet because e.g. seeding")
```
Comment (anonymouse64) [inline boot/seal_test.go:1000]:
```suggestion
		return false, fmt.Errorf("hook not available yet because e.g. seeding")
```

**Review by anonymouse64 — APPROVED** [review]

**Pull request merged**